src/lib/jsonFetcher.js

```diff
@@ -14,73 +19,66 @@ var fetcher = {
-        promise.resolve(data, response.status);
-      }).fail(function(err) {
-        var errorJSON;
-        if (err.responseJSON)
-          errorJSON = err.responseJSON;
-        else if (err.responseText)
-          errorJSON = JSON.parse(err.responseText);
+      .done(function (data, result, response) {
+        promise.resolve(ensureDataIsObject(data), response.status)
+      })
+      .fail(function(error) {
+        var errorJSON
+        if (error.responseJSON)
+          errorJSON = error.responseJSON
+        else if (error.responseText)
+          errorJSON = JSON.parse(error.responseText)
         else
-          errorJSON = {errors: ["We're sorry, an unexpected error has occurred. Please try again; our engineers have been notified."]};
-        promise.reject(errorJSON);
-      });
+          errorJSON = {errors: ["We're sorry, an unexpected error has occurred. Please try again; our engineers have been notified."]}
+        promise.reject(errorJSON)
+      })
 
-    return promise;
+    return promise
   },
-  fetch: function(url) { //temporary backwards compatibility
-    return this.get(url);
+  fetch: function(url) { // temporary backwards compatibility
+    return this.get(url)
   },
   get: function(url, withCredentials) {
-    return this.sendRequest(this.buildRequest(url, 'GET', null, withCredentials));
+    return this.sendRequest(this.buildRequest(url, 'GET', null, withCredentials))
   },
   put: function(url, data) {
-    return this.sendRequest(this.buildRequest(url, 'PUT', data));
+    return this.sendRequest(this.buildRequest(url, 'PUT', data))
   },
   post: function(url, data, withCredentials) {
-    return this.sendRequest(this.buildRequest(url, 'POST', data, withCredentials));
+    return this.sendRequest(this.buildRequest(url, 'POST', data, withCredentials))
   },
-  delete: function(url, data, withCredentials){
-    return this.sendRequest(this.buildRequest(url, 'DELETE', data, withCredentials));
+  delete: function(url, data, withCredentials) {
+    return this.sendRequest(this.buildRequest(url, 'DELETE', data, withCredentials))
   },
   postFile: function(url, data) {
-    //taken from: http://stackoverflow.com/questions/12431760/html5-formdata-file-upload-with-rubyonrails
-    //and http://stackoverflow.com/questions/21234106/upload-file-using-reactjs-via-blueimp-fileupload-jquery-plugin
-    var request = this.buildRequest(url, 'POST', data);
+    // Taken from: http://stackoverflow.com/questions/12431760/html5-formdata-file-upload-with-rubyonrails
+    // and http://stackoverflow.com/questions/21234106/upload-file-using-reactjs-via-blueimp-fileupload-jquery-plugin
+    var request = this.buildRequest(url, 'POST', data)
     request.cache = false
     request.processData = false
     request.contentType = false
-    return this.sendRequest(request);
-  },
-  parseError: function(err) {
```

Comment:
@BJK It was never used or passed anywhere. I figured if we ever need it we can just find it again with `git log -S'parseError'`.